### PR TITLE
Client Side Locale friendly Dates

### PR DESCRIPTION
**Pull request by mcshaz**
This takes all locale time handling from the server to the browser (#188):

1. all times/dates are initially sent to the browser as ISO 8601 strings - therefore if the browser happens to have JS disabled, it will make some sense to the person reading.
2. the gulpfile has been changed to make the relevant time strings from within the .resx files to a javascript object. I have **not** set this up as an automated task yet.
3. The formatting of the dates and times and the language used is taken from browser languages preferences. Even on the same forum I will see dd/mm/yyyy while you will see mm/dd/yyyy.
4. The calculation of UTC to local intervals is client side, and so it will make sense regardless of the timezone of the server and various forum users in different parts of the world. This also has the advantage of keeping ahead of a myriad of daylight savings rules which are ever changing.
5. all AJAX calls regarding date time strings are removed.

### CODE REVIEW

File: src/PopForums.Web/Views/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -1,12 +1,11 @@
 ﻿<!DOCTYPE html>
-<html>
+<html lang="@System.Threading.Thread.CurrentThread.CurrentUICulture.Name">
```
Comment (jeffputz) [inline src/PopForums.Web/Views/Shared/_Layout.cshtml]: This is really scope creep here... a PR shouldn't be reaching into all different kinds of places outside of the original problem domain.

**Comment by jeffputz** [comment]
While I generally appreciate the effort here, as I said in the issue, this might be a lot of work on a dated design that might be entirely replaced.

**Comment by mcshaz** [comment]
Fair call & Apologies for the scope creep.

I fully understand you may not wish to pull the request - particularly as I have broken into a separate ECMAScript class to separate concerns to do with regional time handling (alongside the desire to ring the appropriate components of the resx file into client javascript), which then results in significant changes to gulpfile as well. Having said that this is the engine behind the regional time handling & updating of intervals, so is portable to any rewrite or framework.

I need the changes for personal deployment (we are currently UTC+13 at the moment which is not even selectable, and dates for changing daylight saving are both different dates and different directions). Most people don't even know which time zone they are on as it relates to UTC in my part of the world, and many large cities are not in the list to help users identify the TZ.

Using the lang attribute on the html element was actually a much cleaner way to ensure the correct language is selected for the time since posting to the forum, so this is related. However mia culpa for cleaning up another things which should be in separate pull request, including:
-getting rid of the invalid <div/>
-fix incorrectly redirecting login to the wrong place (account/login rather than forums/account/login)
-serve pre-bundled bootstrap + popper
-serve the .min version of Vue in production. as per the comments in the code, this is actually a different script (not just minified). The only advantage other than loading speed is not having a console with "do not use this script in development" warnings.

as such i'll close the request with my appologies.

I would encourage you at some point to look at what I have done as it allows forum administrators and users not to have to select the timezone they are in (operating systems are really good at this), handles daylight saving incl. legislation changes without updates to PopForums, results in reduced server load (not using AJAX requests when the client can do it) etc.

**Comment by jeffputz** [comment]
@mcshaz No need to apologize! I'm excited that you're willing to put in the work. I'm just not sure if it's compatible with some of the long term goals I have in mind (which are admittedly not well communicated). I think your approach is generally a good idea, so let's keep it up open for now if only for the purpose of looking at what you've done.

**Comment by mcshaz** [comment]
A big change in (seemingly now redundant) server side code has been made on the branch https://github.com/mcshaz/POPForums/tree/remove-serverside-time, and just gone live with the code on https://anzics.nz/Forums. It seems to be working, but I guess I'll find out as my colleagues sign up.

**Comment by mcshaz** [comment]
Sorry - I think I closed this by accident - it is definitely working and worth taking a look at implementing this. The only downside of following through to the other fork at https://github.com/mcshaz/POPForums/tree/remove-serverside-time is you may eventually want the clients timezone for the purpose of emailing client aware dates & times (although you can see how I got around this by purely specifying intervals in the fork [english only] https://github.com/mcshaz/POPForums.Sample/blob/master/src/EmailForumUpdates/Extensions/Dates.cs)
